docs/software/cloud.md

```diff
@@ -110,6 +110,80 @@ Nextcloud gives you access to all your files wherever you are. Store your docume
     ncc [followed by the desired command]
     ```
 
+=== "Nextcloud 'Brute force protection'"
+
+    Nextcloud offers built in brute force protection and additionally a plugin ***Brute-force settings***.  
+    This will delay your login rate in case of several failed login attempts.
+
+    This protection can be extended with Fail2Ban (see following tab).
+
+    See also: <https://apps.nextcloud.com/apps/bruteforcesettings>
+
+=== "Fail2Ban integration"
+
+    Using Fail2Ban your can block users after failed login attempts. This hardens your system, e.g. against brute force attacks.
+
+    - Set options in the ***Nextcloud configuration file*** (typical `/var/www/nextcloud/config/config.php`):
+
+        - Add trusted domains if not already set via the `'trusted_domains'` entry.
+
+            ```ini
+            'trusted_domains' =>
+             array (
+               0 => 'localhost',
+               1 => '<your_trusted_domain>',
+             ),
+            ```
+
+        - Check resp. edit the log file options.  
+          Set the `logtimezone` to the value of your system.  
+          Check whether the `logfile` path matches the entry done in the *Fail2Ban jail file* (`nextcloud.local`, see below).
+
+            ```ini
+            'log_type' => 'file',
+            'logtimezone' => 'Europe/Berlin',
+            'logfile' => '/var/log/nextcloud.log',
+            'loglevel' => 2,
+            ```
+
+    - Create new ***Fail2Ban filter*** (e.g. `/etc/fail2ban/filter.d/nextcloud.conf`):
+
+        ```ini
+        # Fail2Ban filter for Nextcloud
+
+        [Definition]
+        _groupsre = (?:(?:,?\s*"\w+":(?:"[^"]+"|\w+))*)
+        failregex = ^\{%(_groupsre)s,?\s*"remoteAddr":"<HOST>"%(_groupsre)s,?\s*"message":"Login failed:
+                    ^\{%(_groupsre)s,?\s*"remoteAddr":"<HOST>"%(_groupsre)s,?\s*"message":"Trusted domain error.
+        datepattern = ,?\s*"time"\s*:\s*"%%Y-%%m-%%d[T ]%%H:%%M:%%S(%%z)?"
+        ```
+
+    - Create new ***Fail2Ban jail file*** `/etc/fail2ban/jail.d/nextcloud.local`:
+
+        ```ini
+        [nextcloud]
+        backend = auto
+        enabled = true
+        port = http,https
+        protocol = tcp
+        filter = nextcloud
```

Comment:
Btw filter does not need to be set when the filter name matches the jail name, which is reasonably the case here, however, it does not hurt in case the default jail is manipulated.

Reply:
Ok. I think to set these values helps to understand possible settings instead of going the "minimal" path.